File: 1592-rearrange-spaces-between-words/1592-rearrange-spaces-between-words.py
class Solution:
    def reorderSpaces(self, text: str) -> str:
        spaces = text.count(' ')
        words = text.split()
        n = len(words) - 1
        x, y = divmod(spaces, n) if n else (0, spaces)
        
        return (' ' * x).join(words) + ' ' * y
        
        
        
  

File: 1592-rearrange-spaces-between-words/1592-rearrange-spaces-between-words.js
/**
 * @param {string} text
 * @return {string}
 */
var reorderSpaces = function(text) {
  const spaces = text.split('').filter(x => x === ' ').length;
  const words = text.trim().split(/\s+/);
  const n = words.length;
  
  return n === 1
    ? words + ' '.repeat(spaces)
    : words.join(' '.repeat(~~(spaces / (n - 1)))) + ' '.repeat(spaces % (n - 1));
};